packages/components/src/public/QueryModel/QueryModel.ts

```diff
@@ -320,7 +320,8 @@ export class QueryModel {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     readonly rows?: { [key: string]: any };
     /**
-     * The total count of rows for the given QueryModel.
+     * The count of rows returned from the query for the given QueryModel. If includeTotalCount is true, this will
+     * be the total count of rows for the query parameters.
```

Comment:
This seems like it's going to be confusing, even if consistent with what the server does since `rowCount` in the model means different things.  I guess `rows.length` is what should be used when you want to know how many rows there are and `rowCount` only when you want the total.  Would it be helpful to provide different methods on the model to make this distinction more clear? Something like `get totalRowCount` which returns `undefined` if `includeTotalCount` is false and otherwise returns `rowCount`, and `get numRows`, which returns `Object.keys(rows).length` or something long those lines?

Reply:
@labkey-nicka @labkey-susanh @labkey-alan  Yes, that is essentially what I was trying to get at with the rowCount vs totalRows distinction. If you have a rowCount, you really can't know if that was the total rows vs just the ones returned from the query response. I agree that ideally, usages would stop relying on that rowCount when they really should be checking for rows.length. Thoughts? I can definitely add those helper methods if we think that would clarify things.

Reply:
In thinking back to most of the rowCount usages that I reviewed for this feature branch, by far the most common use case was just checking if rowCount === 0 or 1 as a way to check if the query found a row or did not. That type of check really won't differ if the caller uses rows.length or rowCount.